cmd/recv.go

```diff
@@ -199,6 +199,18 @@ func recvAction(cmd *cobra.Command, args []string) {
 					bail("Read zip error: %s", err)
 				}
 
+				// calculate the uncompressed size of the contents of the zip
+				var actualUncompressedSize uint64
+				var fileCount int
+				for _, f := range zr.File {
+					actualUncompressedSize += f.FileHeader.UncompressedSize64
+					fileCount++
+				}
+				if msg.UncompressedBytes64 != int64(actualUncompressedSize) ||
+					msg.FileCount != fileCount {
+					bail("zip error: corrupted zip file")
```

Comment:
Lets change this message to say the zip file didn't match the offer and include exactly what was mismatched. A more specific error message will help if there are buggy clients out there.